src/flask_container_scaffold/util.py

```diff
@@ -45,3 +50,33 @@ def _parse_cfg(config):
             settings_dict[key] = config[section][key]
         config_dict.update({section: settings_dict})
     return config_dict
+
+
+def parse_input(logger, obj, default_return=BaseApiModel):
+    """
+    Parses incoming request, returns a serializable object to return
+    to the client in all cases. When there is a failure, the
+    object contains error information.
+    :param Logger logger: Instantiated logger object
+    :param BaseModel obj: An object type based on a pydantic BaseModel to
+                          attempt to parse.
+    :param BaseApiModel default_return: An object type that will be returned if
+                                        validation of obj fails. This object
+                                        must descend from BaseApiModel or
+                                        implement an error field of type str.
+    :returns: Instantiated object of type obj on success, or default_return
+              on failure to parse.
+    """
+    try:
+        if request.is_json:
+            parsed_args = obj.model_validate_json(json.dumps(request.json))
+        else:
+            if request.args:
+                args = request.args
+            else:
+                args = request.form
+            parsed_args = obj.model_validate_json(json.dumps(args.to_dict()))
+    except ValidationError as e:
+        logger.error(f"Validation error is: {e}")
+        parsed_args = default_return(error=str(e.errors()[0]))
```

Comment:
question: This is possibly a little far-fetched, but what if the default_return class has some required attributes other than 'error'? Do we want to catch a pydantic ValidationError here and fall back to BaseApiModel in that case?

Reply:
This is a good point, I can see this being possible for sure. I don't love the thought of having a try/except _inside_ a try/except, will have to think how that may be handled. Would it be reasonable to make this more resilient in an upcoming patch? This would give us a bit more time to come up with a clean solution to handle this, unless someone suggests something that could be quickly implemented here.

Reply:
You could provide an error handler that is called instead. That would mean adding another parameter to the function, however that would allow you to custom set any data you would want. The default could be code here. This has the advantage of being very flexible at the cost of an additional parameter.